elastic/kibana
# [TSVB] Handle correctly {{key}} placeholder on series name

## Summary

Fixes #80823. On TSVB user can add a `{{key}}` placeholder on series name. This worked only on time series and for group by string/number.
This PR makes it work for:
- timeseries and group by date/boolean
- gauge/goal (didn't work for any type)
- metric (didn't work for any type)

On Metric and Gauge/goal if the {{key}} placeholder is found I display the agg label because I don't think that it makes sense to display the same info on both placeholders (primary and additional label)

### How to test
-  Create a TSVB visualization
- Group by Terms
- Select a string/number/date/boolean field
- Check timeseries, metric, gauge/goal

**Timeseries with boolean field**
<img width="1677" alt="Screenshot 2020-10-27 at 1 33 33 PM" src="https://user-images.githubusercontent.com/17003240/97297302-80aee180-185a-11eb-90f1-ccc076a07c3e.png">

**Timeseries with date**
<img width="1685" alt="Screenshot 2020-10-27 at 1 33 49 PM" src="https://user-images.githubusercontent.com/17003240/97297359-945a4800-185a-11eb-8a8c-a67a1f7fb23b.png">

**Metric**
<img width="1669" alt="Screenshot 2020-10-27 at 1 34 15 PM" src="https://user-images.githubusercontent.com/17003240/97297393-aa680880-185a-11eb-855f-c789e729014c.png">
**Gauge/Goal**
<img width="1679" alt="Screenshot 2020-10-27 at 1 34 25 PM" src="https://user-images.githubusercontent.com/17003240/97297413-b18f1680-185a-11eb-999c-8aeed685f037.png">

### Checklist
